velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -0,0 +1,591 @@
+      auto& writerTyped = writer.castTo<DynamicRow>();
+      if (value.type().error() != ::simdjson::SUCCESS) {
+        writerTyped.set_null_at(0);
+        return simdjson::SUCCESS;
+      }
+      auto type = value.type().value_unsafe();
+      if (type == simdjson::ondemand::json_type::object) {
+        SIMDJSON_ASSIGN_OR_RAISE(auto object, value.get_object());
+
+        folly::F14FastMap<std::string, int32_t> fieldIndices;
+        const auto size = rowType.size();
+        for (auto i = 0; i < size; ++i) {
+          std::string key = rowType.nameOf(i);
+          fieldIndices[key] = i;
+        }
+
+        std::string key;
+        for (auto fieldResult : object) {
+          if (fieldResult.error() != ::simdjson::SUCCESS) {
+            continue;
+          }
+          auto field = fieldResult.value_unsafe();
+          if (!field.value().is_null()) {
+            SIMDJSON_ASSIGN_OR_RAISE(key, field.unescaped_key(true));
+
+            auto it = fieldIndices.find(key);
+            if (it != fieldIndices.end()) {
+              const auto index = it->second;
+              it->second = -1;
+
+              auto res = VELOX_DYNAMIC_TYPE_DISPATCH(
+                  ParseJsonTypedImpl<simdjson::ondemand::value>::apply,
+                  rowType.childAt(index)->kind(),
+                  field.value(),
+                  writerTyped.get_writer_at(index),
+                  false);
+              if (res != simdjson::SUCCESS) {
+                writerTyped.set_null_at(index);
+              }
+            }
+          }
+        }
+
+        for (const auto& [key, index] : fieldIndices) {
+          if (index >= 0) {
+            writerTyped.set_null_at(index);
+          }
+        }
+      } else {
+        if (isRoot) {
+          writerTyped.set_null_at(0);
+          return simdjson::SUCCESS;
+        } else {
+          return simdjson::INCORRECT_TYPE;
+        }
+      }
+      return simdjson::SUCCESS;
+    }
+  };
+
+  static simdjson::simdjson_result<std::string_view> rawJson(
+      Input value,
+      simdjson::ondemand::json_type type) {
+    switch (type) {
+      case simdjson::ondemand::json_type::array: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto array, value.get_array());
+        return array.raw_json();
+      }
+      case simdjson::ondemand::json_type::object: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto object, value.get_object());
+        return object.raw_json();
+      }
+      default:
+        return value.raw_json_token();
+    }
+  }
+
+  template <typename T>
+  static simdjson::error_code castJsonToInt(
+      Input value,
+      exec::GenericWriter& writer) {
+    SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+    switch (type) {
+      case simdjson::ondemand::json_type::number: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto num, value.get_number());
+        switch (num.get_number_type()) {
+          case simdjson::ondemand::number_type::floating_point_number:
+            return simdjson::INCORRECT_TYPE;
```

Comment:
Is there any unhandled number type? Perhaps return 'INCORRECT_TYPE' for the default branch.

Reply:
Thanks, move to default branch. There should be no other numeric types.